homeassistant/components/livisi/switch.py

```diff
@@ -0,0 +1,170 @@
+"""Code to handle a Livisi switches."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DOMAIN,
+    LIVISI_REACHABILITY_CHANGE,
+    LIVISI_STATE_CHANGE,
+    LOGGER,
+    PSS_DEVICE_TYPE,
+)
+from .coordinator import LivisiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch device."""
+    coordinator: LivisiDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add switch."""
+        shc_devices: list[dict[str, Any]] = coordinator.data
+        for device in shc_devices:
+            if device["type"] == PSS_DEVICE_TYPE:
+                if device not in coordinator.switch_devices:
+                    livisi_switch: SwitchEntity = create_entity(
+                        config_entry, device, coordinator
+                    )
+                    LOGGER.debug("Include device type: %s", device.get("type"))
+                    coordinator.switch_devices.append(device)
+                    async_add_entities([livisi_switch])
+
+    coordinator.async_add_listener(handle_coordinator_update)
+
+
+def create_entity(
+    config_entry: ConfigEntry,
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> SwitchEntity:
+    """Create Switch Entity."""
+    config_details: dict[str, Any] = device["config"]
+    capabilities: list = device["capabilities"]
+    room_id: str = device["location"].removeprefix("/location/")
```

Comment:
Why doesn't the device provide the room id by default? Removing a prefix doesn't seem like a proper high level api for the library.

Reply:
I moved the removal of the prefix into the aiolivisi library.